excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/CategoryValidator.java

```diff
@@ -0,0 +1,165 @@
+import com.microsoft.applicationinsights.boot.dependencies.apachecommons.lang3.tuple.Pair;
+import com.microsoft.applicationinsights.boot.dependencies.apachecommons.lang3.tuple.Triple;
+import lombok.val;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.excel.endpoint.exception.InvalidImportException;
+import uk.gov.hmcts.ccd.definition.store.excel.parser.ParseContext;
+import uk.gov.hmcts.ccd.definition.store.excel.parser.model.DefinitionDataItem;
+import uk.gov.hmcts.ccd.definition.store.excel.util.mapper.ColumnName;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+
+@Component
+public class CategoryValidator {
+
+    private ParseContext parseContext;
+    private static final String ERROR_MESSAGE = "CategoryTab Invalid ";
+    private static final String NOT_VALID = " is not a valid ";
+    private static final String DISPLAY_ORDER_ERROR = "DisplayOrder cannot be duplicated within case type"
+        + " category and parent category in Category tab";
+
+
+    public CategoryValidator() {
+    }
+
+    public void validate(ParseContext parseContext, List<DefinitionDataItem> categoryItems) {
+        this.parseContext = parseContext;
+        validateUniqueCategoriesIds(categoryItems);
+        validateUniqueDisplayOrder(categoryItems);
+        val pairCaseTypeCategory = createCategoryIdCaseTypeMap(categoryItems);
+        categoryItems.forEach(element -> validate(element, pairCaseTypeCategory));
+    }
+
+    private void validate(DefinitionDataItem definitionDataItem,Map<String, List<String>> pairCaseTypeCategory) {
+        validateCaseType(definitionDataItem);
+        validateCategoryId(definitionDataItem);
+        validateCategoryLabel(definitionDataItem);
+        validateDisplayOrder(definitionDataItem);
+        validateParentCategoryID(definitionDataItem,pairCaseTypeCategory);
+    }
+
+    private void validateParentCategoryID(DefinitionDataItem definitionDataItem,
+                                          Map<String, List<String>> pairCaseTypeCategory) {
+
+        val parentCategoryId = definitionDataItem.getString(ColumnName.PARENT_CATEGORY_ID);
+        if (!StringUtils.isEmpty(parentCategoryId)) {
+            val error = new InvalidImportException(
+                ERROR_MESSAGE + "ParentCategoryID: " + parentCategoryId + " belongs to a different case type."
+            );
+            val currentCaseType = definitionDataItem.getString(ColumnName.CASE_TYPE_ID);
+            val listOfCategoriesIdsForACaseType = pairCaseTypeCategory.get(currentCaseType);
+
+            if (listOfCategoriesIdsForACaseType.isEmpty() |
+                !listOfCategoriesIdsForACaseType.contains(parentCategoryId)) {
+
+                throw error;
```

Comment:
```suggestion
                throw new InvalidImportException(
                ERROR_MESSAGE + "ParentCategoryID: " + parentCategoryId + " belongs to a different case type."
            );
```
Then we don't need to create the error message unless we are actually throwing the error